maven-scm-api/src/main/java/org/apache/maven/scm/ScmBranchParameters.java

```diff
@@ -34,11 +34,14 @@
 
     private boolean remoteBranching = false;
 
+    private boolean pinExternals = false;
```

Comment:
Is there a usecaes to pin externals with branches?

Reply:
What about the following scenario:
You create a branch for a release version to be able to do special release hotfixes. The externals refer to a different project's trunk that is developed / restructured / whatever in a way that is incompatible with your actual release.
Without pinned externals, your branch won't work anymore.

In that case, yes I think it's useful / necessary to be able to pin externals even in branches.

Reply:
Granted, makes sense!